README.md

```diff
@@ -139,7 +139,41 @@ If `nifi_is_secure` is `True` you must also include
        git_key: # Optional ssh key to use to clone the repository
        destination: # the destination in the Nifi Home folder to copy the downloaded repository to
 
-## Backup / Restore
+    # NiFi parameter contexts
+    nifi_set_parameter_contexts: true
+    nifi_java_home: /usr/lib/jvm/java-8-openjdk-amd64/jre
+    nifi_all_parameter_contexts_result_file: /tmp/all_parameter_contexts_result.json
+    nifi_api_base_url: http://localhost:8080/nifi-api
+    nifi_parameter_contexts:
+
+Setting Parameter Contexts
+--------------------------
+
+You can set NiFi parameter contexts like so:
+
+```yml
+nifi_set_parameter_contexts: true
+nifi_java_home: /usr/lib/jvm/java-8-openjdk-amd64/jre
+nifi_all_parameter_contexts_result_file: /tmp/all_parameter_contexts_result.json
+nifi_api_base_url: http://localhost:8080/nifi-api
+nifi_parameter_contexts:
```

Comment:
I think we can actually simplify some of this by limiting the scope of the tool - IMO we actually don't want to be storing parameter contexts in ansible (those are stored on the nifi server and in backups) - we just want a way of resetting parameter values of *existing* contexts.

Reply:
For this first iteration, I think our "MVP" should be to only move credentials to code, deployed via ansible -- this covers the need for rotation that we have the most demand for.

The use case I’m thinking we're solving here, and I am admittedly a bit lost, is:
- we have our flow credentials in code
- we need to rotate our credentials
- we merge new credentials to playbooks master
- we deploy
- credentials in flow are now updated

Reply:
Actually, maybe there's a hack here that makes everybody happy - we can't embed metadata in parameter names, but we *can* embed metadata in the parameter descriptions.

The idea is to make the logic here dumber - we just read in every parameter of every parameter group, and anything that's tagged with "ansible_managed_by:variable_name" in the description gets replaced by the ansible variable in the tag.

It'll be a lot simpler and less error-prone to add descriptions in NiFi vs updating the playbook for every parameter context change, and it fits with the future vision of parameter contexts to allow referencing other parameter groups: https://cwiki.apache.org/confluence/display/NIFI/Better+Support+for+Parameterizing+Flows - when you can reference other parameter contexts we'd just push all the ansible variables into a root parameter context instead and stop using the description tags.